internal/scm-clients/clients/clients.go

```diff
@@ -87,31 +87,31 @@ func getRepoInfo(repoFullUrl string) (string, string, string, error) {
 	if len(path) < 3 {
 		return "", "", "", fmt.Errorf("missing org/repo in the repository url: %s", repoFullUrl)
 	}
-	repo := path[len(path) - 1]
+	repo := path[len(path)-1]
 	namespace := strings.Split(u.Path, repo)[0]
 	trimedNamespace := namespace[1:(len(namespace) - 1)]
 
 	return u.Host, trimedNamespace, repo, nil
 }
 
-func getClientAdapter(scmName string, accessToken string) (adapter.ClientAdapter, error) {
+func getClientAdapter(scmName string, accessToken string, host string) (adapter.ClientAdapter, error) {
 	var err error
 	var adapter adapter.ClientAdapter
 	httpClient := utils.GetHttpClient(accessToken)
 
 	switch scmName {
```

Comment:
for keeping it an optional feature, you can Init a parameter with the self-hosted-scm value and if its null set the hostname that was extracted from the repo-url